### PR TITLE
tf.range

### PR DESCRIPTION
**Pull request by nbcsm**
support tf.range using loop op

### CODE REVIEW

File: tf2onnx/tfonnx.py [hunk 1]
```diff
@@ -185,14 +185,16 @@ def direct_op(ctx, node, name, args):
...
-        # if identity has a const as input, remove it
+        # if identity has a const as input, remove it if it is not an model output
         input_name = node.input[0]
         output_name = node.output[0]
         for n in ctx.get_nodes():
             for i, parent_name in enumerate(n.input):
                 if parent_name == output_name:
                     n.input[i] = input_name
-        return None
+        if not ctx.is_model_output(node.name):
```
Comment (pengwa) [inline tf2onnx/tfonnx.py]: shall we just use node.output[0] in ctx.output_names?

File: tf2onnx/tfonnx.py [hunk 2]
```diff
@@ -185,18 +187,105 @@ def direct_op(ctx, node, name, args):
...
-        return None
+        if not ctx.is_model_output(output_name):
+            return None
+
     ctx.copy_shape(node.input[0], node.output[0])
     return node
 
 
+def range_op(ctx, node, name, args):
+    """Range."""
```
Comment (pengwa) [inline tf2onnx/tfonnx.py]: list Range signature like this? 

    # T output = Select(bool condition, T x, T y)
    # V v_final_and_scan_outputs = Loop(int64 M, B cond, V v_initial)

File: tf2onnx/tfonnx.py [hunk 3]
```diff
@@ -185,18 +187,105 @@ def direct_op(ctx, node, name, args):
...
+
+    nodes = []
+
+    start_name = start_node.output[0]
+    limit_name = limit_node.output[0]
+    delta_name = delta_node.output[0]
+
+    # trip_count
+    diff_name = "{}_diff".format(base_name)
+    nodes.append(Node(helper.make_node("Sub", [limit_name, start_name], [diff_name], name=diff_name), ctx))
```
Comment (pengwa) [inline tf2onnx/tfonnx.py]: for output id, usually we use port_name(diff_name)

File: tf2onnx/tfonnx.py [hunk 4]
```diff
@@ -1591,6 +1680,7 @@ def mknode(op_type, inputs, opname, **kwargs):
     "RandomUniform": (direct_op, []),
     "RandomNormalLike": (direct_op, []),
     "RandomUniformLike": (direct_op, []),
+    "Range": (range_op, []),
```
Comment (pengwa) [inline tf2onnx/tfonnx.py]: add into opset 7?

File: tf2onnx/tfonnx.py (hunk 3 above)
Comment (nbcsm) [reply tf2onnx/tfonnx.py]: sure

**Pull request merged**